imagetagger/imagetagger/images/management/commands/runzipdaemon.py

```diff
@@ -0,0 +1,48 @@
+import fasteners
+import os
+import sys
+from time import sleep
+
+import zipfile
+from django.conf import settings
+from django.core.management import BaseCommand
+from django.db import transaction
+from django.db.models import Q
+
+from imagetagger.images.models import ImageSet
+
+
+class Command(BaseCommand):
+    help = 'Runs a daemon that creates zip archives for image sets and keeps them up to date'
+
+    def handle(self, *args, **options):
+        if not settings.ALLOW_ZIP_DOWNLOAD:
+            sys.stderr.write('To enable zip download, set ALLOW_ZIP_DOWNLOAD to True in settings.py\n')
+            return
+
+        lock = fasteners.InterProcessLock(os.path.join(settings.IMAGE_PATH, 'zip.lock'))
+        gotten = lock.acquire(blocking=False)
+        if gotten:
+            while True:
+                for imageset in ImageSet.objects.filter(~Q(zip_state=ImageSet.ZipState.READY)):
```

Comment:
I would rather abstract this filter into a model-specific static method as well. Probably the locking bit would live there better.

Reply:
Since the management command is the one that actually zips the files, I think it is okay to lock there.